### ISSUE TITLE
Crash rebinning event workspace to larger bin width

### ISSUE THREAD

**Issue by gemmaguest** [mantidproject/mantid#19096]
This was found while working on #18829 

### Expected behavior
Mantid should not crash

### Actual behavior
Mantid intermittently crashes when rebinning from a large number of bins to a small number of bins

### Steps to reproduce the behavior
- Create a sample workspace with this script:
```
CreateSampleWorkspace(OutputWorkspace='sampleWS', WorkspaceType='Event', NumEvents=100000, XMax=10000, BinWidth=1)
```
- View the workspace and drag the horizontal scrollbar all the way to the right.
- Right-click the workspace in the list and select `Color Fill Plot`
- Move the plot window so that the workspace matrix is fully visible.
- Run `Rebin` with input and output workspaces set to `sampleWS` and params set to `1000`
- Mantid intermittently crashes at this point

The same crash happens with the `ResampleX` algorithm with a small `NumberBins`, e.g. `10`.

### Platforms affected
Maybe all. Seen on Windows 7 with v3.9.0 and v3.9.1-1415-g89802b

**Comment by gemmaguest** [comment]
The problem seems to be that the GUI still thinks there are 10,000 columns in the workspace matrix and has received a notification to update them, but there are now only 10 values in the underlying vector, which it attempts to access out of range.

**Comment by gemmaguest** [comment]
This seems to be fixed in v6.0.0
